tests/PhenX.EntityFrameworkCore.BulkInsert.Tests/TestHelpers.cs

```diff
@@ -0,0 +1,65 @@
+using Microsoft.EntityFrameworkCore;
+
+using PhenX.EntityFrameworkCore.BulkInsert.Enums;
+using PhenX.EntityFrameworkCore.BulkInsert.Extensions;
+using PhenX.EntityFrameworkCore.BulkInsert.Options;
+using PhenX.EntityFrameworkCore.BulkInsert.Tests.DbContext;
+
+using Xunit;
+
+namespace PhenX.EntityFrameworkCore.BulkInsert.Tests;
+
+public enum InsertStrategy
+{
+    Insert,
+    InsertReturn,
+    InsertAsync,
+    InsertReturnAsync
+}
+
+public static class TestHelpers
+{
+    public static async Task<List<T>> InsertWithStrategyAsync<T>(
+        this TestDbContextBase dbContext,
+        InsertStrategy strategy,
+        List<T> entities,
+        Action<BulkInsertOptions>? configure = null,
+        OnConflictOptions<T>? onConflict = null)
+        where T : TestEntityBase
+    {
+        Skip.If(strategy is InsertStrategy.InsertReturn or InsertStrategy.InsertReturnAsync && dbContext.IsProvider(ProviderType.MySql));
+
+        var runId = Guid.NewGuid();
+        if (entities.Any(x => x.TestRun == default))
+        {
+            foreach (var entity in entities)
+            {
+                if (entity.TestRun == default)
+                {
+                    entity.TestRun = runId;
+                }
+            }
+        }
+        else if (entities.Count > 0)
+        {
+            runId = entities[0].TestRun;
+        }
+
+        var actualConfigure = configure ?? (_ => { });
+        switch (strategy)
+        {
+            case InsertStrategy.InsertReturn:
```

Comment:
My thought was to have two enums or two bools, but it works well like this too, as long as there are no more than two parameters